types/wlr_foreign_toplevel_management_v1.c

```diff
@@ -432,6 +462,19 @@ void wlr_foreign_toplevel_handle_v1_destroy(
 
 	wl_list_remove(&toplevel->link);
 
+	/* need to ensure no other toplevels hold a pointer to this one as
+	 * a parent, so that a later call to foreign_toplevel_manager_bind()
+	 * will not result in a segfault */
+	struct wlr_foreign_toplevel_manager_v1 *manager = toplevel->manager;
+	struct wlr_foreign_toplevel_handle_v1 *tl, *tmp3;
+	wl_list_for_each_safe(tl, tmp3, &manager->toplevels, link) {
+		if (tl->parent == toplevel) {
+			/* Note: we do not send a parent signal to the client, that
+			 * should be done by the caller if it wishes */
```

Comment:
Hmm. This won't work because the `parent == toplevel->parent` check in `wlr_foreign_toplevel_handle_v1_set_parent` will bail out early.

I think it'd make sense to just send the event here.